src/keymap.cpp

```diff
@@ -814,19 +812,19 @@ bool KeyMap::is_valid_context(const std::string& context)
 std::map<std::string, Operation> KeyMap::get_internal_operations() const
 {
 	std::map<std::string, Operation> internal_ops;
-	for (int i = 0; opdescs[i].op != OP_NIL; ++i) {
-		if (opdescs[i].flags & KM_INTERNAL) {
-			internal_ops[opdescs[i].default_key] = opdescs[i].op;
+	for (const auto& opdesc : opdescs) {
+		if (opdesc.flags & KM_INTERNAL) {
+			internal_ops[opdesc.default_key] = opdesc.op;
 		}
 	}
 	return internal_ops;
 }
 
 unsigned short KeyMap::get_flag_from_context(const std::string& context)
 {
-	for (unsigned int i = 1; contexts[i] != nullptr; i++) {
-		if (context == contexts[i]) {
-			return (1 << (i - 1)) | KM_SYSKEYS;
+	for (const auto& ctx : contexts) {
```

Comment:
Since `contexts` is a map now, can't we just get the value by key?

Reply:
Jup, good idea.